src/js/components/conversation.jsx

```diff
@@ -62,7 +62,32 @@ export class ConversationComponent extends Component {
     }
 
     render() {
-        const messages = this.props.conversation.messages.map((message) => {
+        const processedMessages = [...this.props.conversation.messages];
+        let lastAuthor;
+        processedMessages.forEach((message, index) => {
+            const author = message.role === 'appUser' ? message.role : `${message.role}_${message.name}`;
+
+            if (!lastAuthor) {
+                lastAuthor = author;
+                message.firstInGroup = true;
```

Comment:
Commenting for both this and https://github.com/smooch/smooch-js/pull/267#discussion_r65170987. I think we should hold a second array in the conversation reducer with either a duplication of the message + those flags, or just the flags and you reconcile them in render. I think this double processing is killing performance in Smoochbot. What do you think @mspensieri ?